### DISCUSSION TITLE
Wasm CommandBar Command Bindings in DataTemplate with ElementName

### DISCUSSION THREAD

**Discussion by e066377** [unoplatform/uno#6600] [Q&A] [answered]
[App1.zip](https://github.com/unoplatform/uno/files/6905693/App1.zip)
Hello,

As my first trial in Uno platform i decided to move an exisitng wpf application for UWP and WASM platforms.
In designing navigation and command binding i encountered some problems in WASM.

I have attached an application showing problems:

1-	CommandBar buttons (AppBarButton) command binding in DataTemplate with ElementName does not work in CommanBar in a DataTemplate for WASM, though they are fine in UWP.
	If AppBarButton is not in CommandBar, then its command binding works even if it is in datatemplate.
	CommandBar buttons command bindings work when it is not in a datatemplate.

	Workaround:
	Setting DataContext of CommandBar and binding commandbar button to the command drectly works as expected.
	<AppBarButton Command="{Binding RefreshCommand}" Label="Refresh" Icon="Refresh" />

2-	The "see more" [...] button of CoomandBar have probles in WASM:
	a) It does not show [...] content and it looks larger.
	b) If selected then CommandBar is disabled completely, the "see more" [...] button and command buttons are not clickable any more.
	c) Command button labels show in this state as expected, but they are not horizontaLly centered as it is in UWP

3-	Tooltips does not work for controls in WASM.
	NavigationView BackButton and menu button show their tooltips as empty rectangles.
	The "see more" [...] button show nothing.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the reports! The CommandBar control is a control that we're currently working on to get in a better shape; it's a custom implementation that is somehow farther from the original implementation in WinUI. We'll have more to share soon that should adress your second point.

The first issue is particularly interesting. It should be working fine and Uno definitely having some trouble finding the `ElementName` in that context. Here's a smaller issue reproducing this: https://github.com/unoplatform/uno/issues/6602

For the last point, the tooltips are showing up properly in my case, though I'm wondering if it could be a language issue. What's the value of `CultureInfo.CurrentCulture` when running in your browser?

**Reply by e066377** [reply]
Yes, for the last part you are right, it was language issue. On my system it was turkish. Setting ApplicationLanguages.PrimaryLanguageOverride = "tr-TR"; solved the problem,  but without setting language there should be a fallback value, like system language or en-Us.

Here https://github.com/unoplatform/uno/issues/6602 it will be better explained if condition CommandBar is in DataTemplate is added.  CommandBar AppBarButton command binding with ElementName as Content of NavigationView works, but not when it is in HeaderDataTemplate.

Thanks.

**Reply by jeromelaban** [reply]
For the language, it's likely related to this issue #6469.
